fmriprep/workflows/bold/base.py

```diff
@@ -689,50 +710,104 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
 *Grayordinates* files [@hcppipelines], which combine surface-sampled
 data and volume-sampled data, were also generated.
 """
-            gen_cifti = pe.MapNode(GenerateCifti(), iterfield=["surface_target", "gifti_files"],
-                                   name="gen_cifti")
-            gen_cifti.inputs.TR = metadata.get("RepetitionTime")
-
-            workflow.connect([
-                (bold_surf_wf, gen_cifti, [
-                    ('targets.out', 'surface_target'),
-                    ('outputnode.surfaces', 'gifti_files')]),
-                (inputnode, gen_cifti, [('subjects_dir', 'subjects_dir')]),
-                (bold_mni_trans_wf, gen_cifti, [('outputnode.bold_mni', 'bold_file')]),
-                (gen_cifti, outputnode, [('out_file', 'bold_cifti'),
-                                         ('variant', 'cifti_variant'),
-                                         ('variant_key', 'cifti_variant_key')]),
-            ])
+                gen_cifti = pe.MapNode(GenerateCifti(),
+                                       iterfield=["surface_target", "gifti_files"],
+                                       name="gen_cifti")
+                gen_cifti.inputs.TR = metadata.get("RepetitionTime")
+
+                workflow.connect([
+                    (bold_surf_denoised_wf, gen_cifti, [
+                        ('targets.out', 'surface_target'),
+                        ('outputnode.surfaces', 'gifti_files')]),
+                    (inputnode, gen_cifti, [('subjects_dir', 'subjects_dir')]),
+                    (bold_mni_trans_wf, gen_cifti, [('outputnode.bold_mni', 'bold_file')]),
+                    (gen_cifti, outputnode, [('out_file', 'bold_cifti'),
+                                             ('variant', 'cifti_variant'),
+                                             ('variant_key', 'cifti_variant_key')]),
+                ])
+            else:
+                bold_surf_wf.__desc__ += """\
+*Grayordinates* files [@hcppipelines], which combine surface-sampled
+data and volume-sampled data, were also generated.
+"""
+                gen_cifti = pe.MapNode(GenerateCifti(),
+                                       iterfield=["surface_target", "gifti_files"],
+                                       name="gen_cifti")
+                gen_cifti.inputs.TR = metadata.get("RepetitionTime")
+
+                workflow.connect([
+                    (bold_surf_wf, gen_cifti, [
+                        ('targets.out', 'surface_target'),
+                        ('outputnode.surfaces', 'gifti_files')]),
+                    (inputnode, gen_cifti, [('subjects_dir', 'subjects_dir')]),
+                    (bold_mni_trans_wf, gen_cifti, [('outputnode.bold_mni', 'bold_file')]),
+                    (gen_cifti, outputnode, [('out_file', 'bold_cifti'),
+                                             ('variant', 'cifti_variant'),
+                                             ('variant_key', 'cifti_variant_key')]),
+                ])
```

Comment:
Almost all of this is the same in both the `if` and `else` block. You could keep the original and change one line:

```Diff
  workflow.connect([
-     (bold_surf_wf, gen_cifti [
+     (bold_surf_denoised_wf if use_aroma else bold_surf_wf, gen_cifti [
          ('targets.out', 'surface_target'),
          ('outputnode.surfaces', 'gifti_files')]),
```